### DISCUSSION TITLE
Built-in type aliases

### DISCUSSION THREAD

**Discussion by data-man** [c3lang/c3c#434] [Polls]
Of course, these types can be defined in stdint(stdfloat/stdtypes/etc.).c3.
But maybe add them to built-in types?

**Comment by lerno** [comment]
Actually, the only way to add the types with such names would be through adding them to the reserved names, which I am not keen on. This would just add another difference between libraries, some libraries that adhere to char/int/long style and others that use bit named types. I originally thought to add all of those aliases (it used to be in the docs), but the problem with different styles and annoying mixed usage made me remove them. It's one or the other.

**Reply by data-man** [reply]
> some libraries that adhere to char/int/long style

When I need a 32/64-bit type I want to make sure that its size does not depend on the platform.

**Reply by lerno** [reply]
In C3, type sized are fixed:

ichar/char 8 bits
short/ushort 16
int/uint 32
long/ulong 64
int128/uint128 128

Variable sized:
ireg/ureg - register sized
usize/isize - size of size_t
iptr/uptr - size of void*
iptrdiff/uptrdiff - size of ptrdiff_t

**Reply by data-man** [reply]
> In C3, type sized are fixed

It's omitted in `In what ways do C3 differ from C?`!
But int128/u128 are certainly different from other names.

**Comment by lerno** [comment]
In addition, `i32` style is objectively bad, as an example: `for (i32 i = 32; i > 0; i--)` this makes readability really poor with such common variable names. `s32/u32` would be the only acceptable single letter prefix scheme. Compare:
```c
for (i32 i = 32; i > 0; i--) { ... }

for (s32 i = 32; i > 0; i--) { ... }
```

**Reply by data-man** [reply]
> makes readability really poor with such common variable names.

Any good editor will highlight built-in types with a other color/style.

> s32/u32 would be the only acceptable single letter prefix scheme.

```rust
for (s32 s = 32; s > 0; s--) { ... }
for (u32 u = 32; u > 0; u--) { ... }
```
:smile:

How often is `i = 32; i > 0` used?

**Reply by lerno** [reply]
You get a similar visual issue with `for (i32 i = 0; i < len; i++)` but just not as strong.

**Comment by data-man** [comment]
Also would be nice to have `wchar/dchar == ushort/uint` (like D language) and/or `char16/char32`.

**Comment by vssukharev** [comment]
Seems like a rust vibe. Personally, I prefer int, uint, ... They have fixed sizes and everything can be checked in docs or manually (I mean their sizes).
